src/nodedef.cpp

```diff
@@ -48,20 +49,32 @@ void NodeBox::reset()
 	wall_top = aabb3f(-BS/2, BS/2-BS/16., -BS/2, BS/2, BS/2, BS/2);
 	wall_bottom = aabb3f(-BS/2, -BS/2, -BS/2, BS/2, -BS/2+BS/16., BS/2);
 	wall_side = aabb3f(-BS/2, -BS/2, -BS/2, -BS/2+BS/16., BS/2, BS/2);
+	// no default for other parts
+	connect_up.clear();
+	connect_down.clear();
+	connect_n.clear();
+	connect_e.clear();
+	connect_w.clear();
+	connect_s.clear();
 }
 
 void NodeBox::serialize(std::ostream &os, u16 protocol_version) const
 {
-	int version = protocol_version >= 21 ? 2 : 1;
+	int version = 1;
+	if (protocol_version >= 27)
+		version = 3;
+	else if (protocol_version >= 21)
+		version = 2;
 	writeU8(os, version);
 
 	if (version == 1 && type == NODEBOX_LEVELED)
 		writeU8(os, NODEBOX_FIXED);
+	else if (version == 2 && type == NODEBOX_CONNECTED)
```

Comment:
Agree with @sofar here, readability is better served if this is separate.

@sofar your suggestion is fine, as long as it doesn't turn into a rewrite of the method...